static/extensions/PuzzlingGGG/underspeech.js

```diff
@@ -0,0 +1,196 @@
+    }
+    this.dialogueText = '';
+    this.timeout = null;
+}
+
+    getInfo() {
+        return {
+            id: 'puzzlingunderspeech',
+            name: 'Underspeech',
+            blocks: [
+                {
+                    opcode: 'showDialogueFromJSON',
+                    blockType: Scratch.BlockType.COMMAND,
+                    text: 'say [JSON] by [SPEAKER] as [TYPE]',
+                    arguments: {
+                        JSON: {
+                            type: Scratch.ArgumentType.STRING,
+                            defaultValue: '{"line": [{"text": "Hello, ", "style": {"font": "Arial", "size": "14px", "color": "black"}}]}'
+                        },
+                        SPEAKER: {
+                            type: Scratch.ArgumentType.STRING,
+                            defaultValue: "Dango"
+                        },
+                        TYPE: {
+                            type: Scratch.ArgumentType.STRING,
+                            menu: 'typeMenu',
+                            defaultValue: 'normal'
+                        },
+                    }
+                },
+                {
+                    opcode: 'hideDialogue',
+                    blockType: Scratch.BlockType.COMMAND,
+                    text: 'hide dialogue'
+                },
+                '---','---','---',
+                {
+                    func: 'openDialogueEditor',
+                    blockType: Scratch.BlockType.BUTTON,
+                    text: 'Open Dialogue Editor'
+                },
+                {
+                    func: 'showPopup',
+                    blockType: Scratch.BlockType.BUTTON,
+                    text: 'Show Examples'
+                }
+            ],
+            menus: {
+                typeMenu: {
+                    acceptReporters: false,
+                    items: ['normal', 'typewriter']
+                }
+            }
+        };
+    }
+
+  createDialogueContainer() {
+    let container = document.createElement('div');
+    container.style.position = 'absolute';
+    container.style.bottom = '0';
+    container.style.left = '0';
+    container.style.width = '100%';
+    container.style.backgroundColor = 'rgba(0, 0, 0, 0.8)';
+    container.style.color = 'white';
+    container.style.textAlign = 'left';
+    container.style.padding = '10px';
+    container.style.boxSizing = 'border-box';
+    container.style.zIndex = '100';
+    container.style.fontFamily = '"Helvetica Neue", Helvetica, Arial, sans-serif';
+    container.style.display = 'none';
+    container.style.pointerEvents = 'none';
+    return container;
+  }
+
+showDialogueFromJSON(args, util) {
+  this.dialogueContainer.innerHTML = '';
+  this.dialogueContainer.style.display = 'block';
+
+  try {
+    const dialogueData = JSON.parse(args.JSON);
+    const speakerHTML = `<div style="font-size: 1.2em; margin-bottom: 0.5em;"><strong>${args.SPEAKER}:</strong></div>`;
```

Comment:
"args.SPEAKER" is dangerous, people can input HTML. Please use some kind of XML escape

Reply:
oh yeah that is dangerous. do args.SPEAKER.replaceAll("&", "&amp;").replaceAll(" ", "&nbsp;").replaceAll("<", "&lt;").replaceAll(">", "&gt;")

Reply:
alternatively you can form the text into the element classes then assign innerText which will automatically escape the text for you